### PR TITLE
add column mapping for batch write

### PR DESCRIPTION
**Pull request by marsishandsome**

### CODE REVIEW

**Comment by marsishandsome** [comment]
@zhexuany PTAL

**Comment by marsishandsome** [comment]
@zhexuany PTAL

File: core/src/main/scala/com/pingcap/tispark/TiBatchWrite.scala [hunk 1]
```diff
@@ -98,10 +102,30 @@ class TiBatchWrite(@transient val df: DataFrame,
...
     }
+    dfColSize = df.columns.length
+    tableColSize = tiTableInfo.getColumns.size()
 
     // check check check
     checkUnsupported()
     checkColumnNumbers()
+
+    // initialize dfTiColumnInfo
+    dfTiColumnInfo = new Array[TiColumnInfo](dfColSize)
```
Comment (zhexuany) [inline core/src/main/scala/com/pingcap/tispark/TiBatchWrite.scala]: this should move to sparkRow2TiRow. Otherwise, we do not have a way to allocate id.

**Pull request merged**